### ISSUE TITLE
Does not output reference when values is zero

### ISSUE THREAD

**Issue by lmacedotzt** [amzn/style-dictionary#986]
When using `outputReference`, if the referenced alias has a value of zero it being ignored and the reference is not being replaced.

Input to recreate the issue:
```code
// json
{
  "Offset": {
    "value": "0",
    "type": "other"
  },
  "Spacing": {
    "value": "2 + {Offset}",
    "type": "spacing"
  }
}
```
Actual result
```code
// css
--offset: 0;
--spacing: 2 + 0;
```

Expected result
```code
// css
--offset: 0;
--spacing: 2 + var(--offset);
```

After a quick look, it seems the issue is here, the condition is checking for a truthy value and zero is falsy: https://github.com/amzn/style-dictionary/blob/2cf72f3f89996503608c238d7f3bcbab5b53e719/lib/common/formatHelpers/createPropertyFormatter.js#LL114C21-L114C21

And I think this issue is related #886

**Comment by acidlynx** [comment]
Like [here](https://github.com/amzn/style-dictionary/issues/988) you can use function in `filter`

```
function(token) {
            // omit empty or zero tokens
            if (token.value !=0) {
              return token
            }
```

**Comment by jorenbroekema** [comment]
I have reproduced this bug. There have been some fixes to outputReferences in `v3.9.0` and `v4.0.0-prerelease.1` which solve the following:
```json
{
  "Offset": {
    "value": "0",
    "type": "other"
  },
  "Spacing": {
    "value": "2 + {Offset}",
    "type": "spacing"
  }
}
```

But when you mentioned the truthy check, I realized there is another bug indeed that is not fixed, which can be reproduced by making the "0" in your JSON a number rather than string:

```json
{
  "Offset": {
    "value": 0,
    "type": "other"
  },
  "Spacing": {
    "value": "2 + {Offset}",
    "type": "spacing"
  }
}
```
